compiler/rustc_codegen_llvm/src/intrinsic_util.rs

```diff
@@ -146,14 +146,14 @@ fn codegen_collectable_calls_for_enum<'a, 'll, 'tcx>(
     };
 
     // Build the successor block.
-    let name = format!("successor_{:?}", adt_def.did.index);
+    let name = format!("successor_{:?}", adt_def.did);
     let successor = builder.build_sibling_block(&name);
 
     // Codegen a basic block for each enum variant. Variants with multiple
     // fields use the same variant block.
     let mut variants = Vec::with_capacity(adt_def.variants.len());
     for (idx, discr) in adt_def.discriminants(builder.tcx) {
-        let name = format!("adt_{:?}_variant_{}", adt_def.did.index, discr.val);
+        let name = format!("adt_{:?}_variant_{}", adt_def.did, discr.val);
```

Comment:
Does `discr.val` guarantee uniqueness? [I haven't got a clue!]

Reply:
Yes, this represents the tag value in the enum.